### PR TITLE
Use Horologist for Scaffold and ScalingLazyColumn

### PR DESCRIPTION
**Pull request by kul3r4**
Migrate the code to use Horologist for Scaffold and ScalingLazyColumn.
This code change requires changes in the codelab itself so do not submit once approved.

### CODE REVIEW

File: finished/src/main/java/com/example/android/wearable/composeforwearos/MainActivity.kt [hunk 1]
```diff
@@ -63,61 +58,56 @@ class MainActivity : ComponentActivity() {
...
     WearAppTheme {
-        // TODO: Swap to ScalingLazyListState
-        val listState = rememberScalingLazyListState()
+        /*
+         * [Horologist] AppScaffold adds a TimeText by default that can be override by the
+         * ScreenScaffold. It ensures that TimeText animates correctly when navigating between
+         * screens.
+         * */
+        AppScaffold {
+            // TODO: Swap to ScalingLazyListState
```
Comment (yschimke) [inline finished/src/main/java/com/example/android/wearable/composeforwearos/MainActivity.kt:71]: remove TODO?

File: finished/src/main/java/com/example/android/wearable/composeforwearos/MainActivity.kt [hunk 2]
```diff
@@ -63,61 +58,56 @@ class MainActivity : ComponentActivity() {
...
 }
 
+@OptIn(ExperimentalHorologistApi::class)
 @Composable
 fun WearApp() {
     WearAppTheme {
-        // TODO: Swap to ScalingLazyListState
-        val listState = rememberScalingLazyListState()
+        /*
+         * [Horologist] AppScaffold adds a TimeText by default that can be override by the
```
Comment (yschimke) [inline finished/src/main/java/com/example/android/wearable/composeforwearos/MainActivity.kt:66]:
```suggestion
         * [Horologist] AppScaffold adds a TimeText by default that can be overriden by the
```

File: finished/src/main/java/com/example/android/wearable/composeforwearos/MainActivity.kt [hunk 3]
```diff
@@ -63,61 +58,56 @@ class MainActivity : ComponentActivity() {
...
+        AppScaffold {
+            // TODO: Swap to ScalingLazyListState
+        /*
+         * Specifying the types of items that appear at the start and end of the list ensures that the
+         * appropriate padding is used.
+         */
+            val listState = rememberResponsiveColumnState(
+                contentPadding = ScalingLazyColumnDefaults.padding(
+                    first = ItemType.Text,
+                    last = ItemType.SingleButton,
```
Comment (yschimke) [inline finished/src/main/java/com/example/android/wearable/composeforwearos/MainActivity.kt:79]: I think you have a Chip?

File: finished/src/main/java/com/example/android/wearable/composeforwearos/MainActivity.kt [hunk 4]
```diff
@@ -63,61 +58,56 @@ class MainActivity : ComponentActivity() {
...
-                /* ******************* Part 1: Simple composables ******************* */
-                item { ButtonExample(contentModifier, iconModifier) }
-                item { TextExample(contentModifier) }
-                item { CardExample(contentModifier, iconModifier) }
+                /* *************************** Part 3: ScalingLazyColumn *************************** */
+                // TODO: Swap a ScalingLazyColumn (Wear's version of LazyColumn)
+                ScalingLazyColumn(
+                    columnState = listState,
+                    modifier = Modifier
+                        .fillMaxSize(),
```
Comment (yschimke) [inline finished/src/main/java/com/example/android/wearable/composeforwearos/MainActivity.kt:97]: not needed

File: finished/src/main/java/com/example/android/wearable/composeforwearos/MainActivity.kt [hunk 5]
```diff
@@ -63,61 +58,56 @@ class MainActivity : ComponentActivity() {
...
+             * */
+            ScreenScaffold(
+                scrollState = listState,
             ) {
-                /* ******************* Part 1: Simple composables ******************* */
-                item { ButtonExample(contentModifier, iconModifier) }
-                item { TextExample(contentModifier) }
-                item { CardExample(contentModifier, iconModifier) }
+                /* *************************** Part 3: ScalingLazyColumn *************************** */
+                // TODO: Swap a ScalingLazyColumn (Wear's version of LazyColumn)
```
Comment (yschimke) [inline finished/src/main/java/com/example/android/wearable/composeforwearos/MainActivity.kt:93]: remove comment? or is it by design?

**Review by yschimke — APPROVED** [review]

File: finished/src/main/java/com/example/android/wearable/composeforwearos/MainActivity.kt (hunk 5 above)
Comment (kul3r4) [reply finished/src/main/java/com/example/android/wearable/composeforwearos/MainActivity.kt:93]: yep by design

File: finished/src/main/java/com/example/android/wearable/composeforwearos/MainActivity.kt (hunk 3 above)
Comment (kul3r4) [reply finished/src/main/java/com/example/android/wearable/composeforwearos/MainActivity.kt]: yep and also the first one is a button, fixed

File: finished/src/main/java/com/example/android/wearable/composeforwearos/MainActivity.kt (hunk 4 above)
Comment (kul3r4) [reply finished/src/main/java/com/example/android/wearable/composeforwearos/MainActivity.kt]: removed. Also added comment for Horologist ScalingLazyColumn

File: finished/src/main/java/com/example/android/wearable/composeforwearos/MainActivity.kt (hunk 1 above)
Comment (kul3r4) [reply finished/src/main/java/com/example/android/wearable/composeforwearos/MainActivity.kt:71]: By codelab design

**Pull request merged**